### PR TITLE
davidsp/readme

### PR DESCRIPTION
**Pull request by dsp-ant**
- **Update README**
- **Update supported language**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,27 @@
 # MCP Python SDK
+[![PyPI][pypi-badge]][pypi-url]
+[![MIT licensed][mit-badge]][mit-url]
+[![Python Version][python-badge]][python-url]
+[![Documentation][docs-badge]][docs-url]
+[![Specification][spec-badge]][spec-url]
+[![GitHub Discussions][discussions-badge]][discussions-url]
 
+<<<<<<< Updated upstream
+=======
```
Comment (jspahrsummers) [inline README.md]: Merge conflict marker

File: README.md [hunk 2]
```diff
@@ -1,5 +1,27 @@
 # MCP Python SDK
+[![PyPI][pypi-badge]][pypi-url]
+[![MIT licensed][mit-badge]][mit-url]
+[![Python Version][python-badge]][python-url]
+[![Documentation][docs-badge]][docs-url]
+[![Specification][spec-badge]][spec-url]
+[![GitHub Discussions][discussions-badge]][discussions-url]
```
Comment (jspahrsummers) [inline README.md:7]: That's a lot of badges! Are these actually that helpful?

File: README.md [hunk 3]
```diff
@@ -13,60 +35,290 @@ The Model Context Protocol allows applications to provide context for LLMs in a
...
 ```
 
+Alternatively, add mcp to your `requirements.txt`:
+```
+pip install mcp
+# or add to requirements.txt
+pip install -r requirements.txt
+```
+
+## Overview
```
Comment (jspahrsummers) [inline README.md:48]: Kinda think we should just leave these bits to the docs. Putting it here just means more places to update, right?

File: README.md [hunk 4]
```diff
@@ -13,60 +35,290 @@ The Model Context Protocol allows applications to provide context for LLMs in a
...
+
+        # Call a tool
+        result = await session.call_tool("tool-name", arguments={"arg1": "value"})
+
+        # Get a prompt
+        prompt = await session.get_prompt("prompt-name", arguments={"arg1": "value"})
 ```
 
-### Creating a Server
+## Primitives
```
Comment (jspahrsummers) [inline README.md:178]: As above.

File: README.md [hunk 5]
```diff
@@ -13,60 +35,290 @@ The Model Context Protocol allows applications to provide context for LLMs in a
...
+
+    return [types.TextContent(type="text", text="Operation complete")]
 ```
 
+The request context is automatically set for each request and provides a safe way to access the current client session and request metadata.
+
 ## Documentation
 
 - [Model Context Protocol documentation](https://modelcontextprotocol.io)
+<<<<<<< Updated upstream
```
Comment (jspahrsummers) [inline README.md]: Merge conflict marker

File: README.md (hunk 2 above)
Comment (dsp-ant) [reply README.md:7]: Yes. I mostly took those from uv and tokio. Seems pretty standard. I can remove the spec or discusison ones.

File: README.md (hunk 3 above)
Comment (dsp-ant) [reply README.md:48]: Yes, but also people look a lot at readmes as the main "how to get started". If you look at tokio.rs, fastapi, or uv, all their readmes are quite extensive.

Comment (jspahrsummers) [reply README.md:48]: I meant more the _conceptual_ docs here. "How to get started" stuff is totally fine, of course.

Comment (dsp-ant) [reply README.md:48]: I think those are actually fine. They are very minimal but quite helpful to see what decorators you have available, what primitives exist, and what the general gist of this is. I think i'd like to keep them for now.

**Pull request merged**